packages/data-context/src/DataContext.ts

```diff
@@ -352,36 +367,60 @@ export class DataContext {
     console.error(e)
   }
 
-  onError = (cypressError: CypressError, title?: string) => {
+  onError = (cypressError: CypressError, title: string = 'Unexpected Error') => {
     if (this.isRunMode) {
       if (this.lifecycleManager?.runModeExitEarly) {
         this.lifecycleManager.runModeExitEarly(cypressError)
       } else {
         throw cypressError
       }
     } else {
-      this.update((coreData) => {
-        coreData.baseError = { title, cypressError }
-      })
+      const err = {
+        id: _.uniqueId('Error'),
+        title,
+        cypressError,
+      }
 
-      this.emitter.baseErrorChange()
+      this.update((d) => {
+        if (d.currentProjectData?.testingTypeData?.activeAppData) {
+          d.currentProjectData.testingTypeData.activeAppData.error = err
+        } else if (d.currentProjectData?.testingTypeData) {
+          d.currentProjectData.testingTypeData.error = err
+        } else if (d.currentProjectData) {
+          d.currentProjectData.error = err
+        } else {
+          d.baseError = err
+        }
+      })
 
-      this.emitter.toLaunchpad()
-      this.emitter.toApp()
+      this.emitter.errorWarningChange()
     }
   }
 
-  onWarning = (err: CypressError) => {
+  onWarning = (err: CypressError, target?: 'testingType' | 'project') => {
```

Comment:
I see one instance of `target` being specified, but I removed it and didn't see any differences. Since we send all the warnings in an array, is this needed?

Reply:
Yeah it actually looking to guard that it doesn't get cleared if it wasn't supposed to, but agree it seems unneeded. Will remove